src/test/java/de/kherud/llama/LlamaModelIT.java

```diff
@@ -86,7 +86,8 @@ public void testGenerateGrammar() {
 		String output = sb.toString();
 
 		Assert.assertTrue(output.matches("[ab]+"));
-		Assert.assertEquals(nPredict, model.encode(output).length);
+		int generated = model.encode(output).length;
+		Assert.assertTrue(generated > 0 && generated <= nPredict);
```

Comment:
I suppose the number of tokens generated parameter is an _upper bound_ rather than a guarantee.  It's consistently off by 1 in the 2 bit quantization.